tls/s2n_connection.c

```diff
@@ -1093,6 +1093,35 @@ uint64_t s2n_connection_get_delay(struct s2n_connection *conn)
     return conn->delay - elapsed;
 }
 
+static int s2n_connection_kill(struct s2n_connection *conn)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_GUARD_RESULT(s2n_connection_set_closed(conn));
+
+    /* Delay between 10 and 30 seconds in nanoseconds */
+    int64_t min = TEN_S, max = 3 * TEN_S;
+
+    /* Keep track of the delay so that it can be enforced */
+    uint64_t rand_delay = 0;
+    POSIX_GUARD_RESULT(s2n_public_random(max - min, &rand_delay));
+
+    conn->delay = min + rand_delay;
+
+    /* Restart the write timer */
+    POSIX_GUARD_RESULT(s2n_timer_start(conn->config, &conn->write_timer));
+
+    if (conn->blinding == S2N_BUILT_IN_BLINDING) {
+        struct timespec sleep_time = { .tv_sec = conn->delay / ONE_S, .tv_nsec = conn->delay % ONE_S };
+        int r;
```

Comment:
nit

```suggestion
        int r = 0;
```

Reply:
Guess I might as well swap it to S2N_RESULT too while I'm touching it :)